atlasdb-config/src/main/java/com/palantir/atlasdb/factory/KvsBackoff.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.factory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KvsBackoff {
+    private static final Logger log = LoggerFactory.getLogger(KvsBackoff.class);
+
+    private static final double GOLDEN_RATIO = (Math.sqrt(5) + 1.0) / 2.0;
+    private static final int QUICK_MAX_BACKOFF_SECONDS = 1000;
+
+    public static void pauseForBackOff(int failureCount) {
+        long timeoutInSeconds = Math.min(QUICK_MAX_BACKOFF_SECONDS, Math.round(Math.pow(GOLDEN_RATIO, failureCount)));
```

Comment:
(just looking at random PRs, feel free to ignore me)

I think this is correct but I had to read docs on Math.round() to see how it behaves if the argument doesn't fit in a long.

It could be more straightforward to make this class instantiable and stateful instead of passing the failure count and dealing with floating point arithmetic (it could just generate fibonacci numbers). Also, instead of sleeping in the class itself, you could return a number of milliseconds to sleep -- that would make it possible to test the class easily if you wanted to.

So the class becomes like this:
```
public class FinbonacciBackoffSequence {
    private int prevPrev = 0;
    private int prev = 1;

    public long getNextBackoffMillis() {
         int next = Math.min(MAX_BACKOFF_SECONDS, prevPrev + prev);
         prevPrev = prev;
         prev = next;
         return 1000 * next;
    }
}
```
At this point, you should probably think that there must be a library for this. And indeed, there is https://github.com/rholder/guava-retrying that we already happen to depend on in `timelock-server`. (That said, I'm not entirely sure if it's justified to add a dependency here)